client/src/components/overview/ProductOverview.jsx

```diff
@@ -8,16 +8,19 @@ import ProductDescription from './ProductDescription';
 import './productOverview.css';
 
 class ProductOverview extends React.Component {
-  constructor(props) {
+  constructor(props) { // = {
+  //   selectedProduct,
+  //   checkCache,
+  //   updateCache
+  // }
     super(props);
 
-    this.numberOfReviews = 0;
-    this.averageRating = 0;
-    this.styles = [];
-
     this.state = {
+      styles: [],
```

Comment:
I would think more about which of these actually need to be tracked, and how well-suited they are to being tracked as component state.

Reply:
Agreed. Just getting it to work for now. Before, styles was a local props, but moving the data to cache requires some rethinking of props vs state & where it needs to be.